quickstart/201-azfw-with-ipgroups/main.tf

```diff
@@ -0,0 +1,273 @@
+  address_prefixes     = ["10.10.2.0/24"]
+}
+
+resource "azurerm_public_ip" "vm_jump_pip" {
+  name                = "pip-jump"
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+  allocation_method   = "Static"
+  sku                 = "Standard"
+}
+
+resource "azurerm_network_interface" "vm_server_nic" {
+  name                = "nic-server"
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+
+  ip_configuration {
+    name                          = "ipconfig-workload"
+    subnet_id                     = azurerm_subnet.server_subnet.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_network_interface" "vm_jump_nic" {
+  name                = "nic-jump"
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+
+  ip_configuration {
+    name                          = "ipconfig-jump"
+    subnet_id                     = azurerm_subnet.jump_subnet.id
+    private_ip_address_allocation = "Dynamic"
+    public_ip_address_id          = azurerm_public_ip.vm_jump_pip.id
+  }
+}
+
+resource "azurerm_network_security_group" "vm_server_nsg" {
+  name                = "nsg-server"
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+}
+
+resource "azurerm_network_security_group" "vm_jump_nsg" {
+  name                = "nsg-jump"
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+  security_rule {
+    name                       = "Allow-TCP"
+    priority                   = 1000
+    direction                  = "Inbound"
+    access                     = "Allow"
+    protocol                   = "Tcp"
+    source_port_range          = "*"
+    destination_port_range     = "22"
+    source_address_prefix      = "*"
+    destination_address_prefix = "*"
+  }
+}
+
+resource "azurerm_network_interface_security_group_association" "vm_server_nsg_association" {
+  network_interface_id      = azurerm_network_interface.vm_server_nic.id
+  network_security_group_id = azurerm_network_security_group.vm_server_nsg.id
+}
+
+resource "azurerm_network_interface_security_group_association" "vm_jump_nsg_association" {
+  network_interface_id      = azurerm_network_interface.vm_jump_nic.id
+  network_security_group_id = azurerm_network_security_group.vm_jump_nsg.id
+}
+
+resource "azurerm_linux_virtual_machine" "vm_server" {
+  name                            = "server-vm"
+  resource_group_name             = azurerm_resource_group.rg.name
+  location                        = azurerm_resource_group.rg.location
+  size                            = var.virtual_machine_size
+  admin_username                  = var.admin_username
+  admin_password                  = random_password.password.result
```

Comment:
Could we use ssh key like this [example](https://github.com/Azure/terraform/blob/master/quickstart/101-vm-cluster-linux/ssh.tf) and it's [reference](https://github.com/Azure/terraform/blob/master/quickstart/101-vm-cluster-linux/main.tf#L103-L106) instead of password?

Reply:
@lonegunmanb Great call out.

@cshea15 The following example shows how we use SSH in our samples: https://github.com/Azure/terraform/tree/master/quickstart/201-k8s-cluster-with-tf-and-aks

Reply:
the reason I went with password was because the other examples for bicep and arm were using password so was trying to keep it the same.

Reply:
@lonegunmanb @TomArcherMsft just looking for a follow up. Is it ok that I keep the password to follow the consistency with the other quickstart or is it required that I should use the SSH Key?

Reply:
@cshea15 Since this sample uses a Linux VM, SSH is best-practice. 
In the reference doc for [azurerm_virtual_machine](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/virtual_machine), please refer to the note on "SSH Key authentication".

For Windows VMs (used in the other two PRs), you should use a password as [azurerm_windows_virtual_machine](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/windows_virtual_machine.html) has no SSH fields. 

@lonegunmanb What is your opinion on this?